src/metabase/notification/send.clj

```diff
@@ -388,14 +390,47 @@
   "The default options for sending a notification."
   {:notification/sync? false})
 
+(setting/defsetting notification-cutoff-timestamp
+  "Timestamp that serves as an anchor point for notifications.
+  Timestamp should be ISO 8601 format and in UTC.
+  Tip: Use the `date -u -Iseconds` command to get the current time in UTC.
+
+  Used for staging instances to skip sending existing notifications."
+  :type       :string
+  :default    nil
+  :encryption :no
+  :export?    false
+  :cache? false
+  :getter     (fn []
+                (when-let [timestamp (some-> (setting/get-value-of-type :string :notification-cutoff-timestamp)
+                                             u.date/parse ;; expects ISO 8601 format
+                                             (#(.toOffsetDateTime ^ZonedDateTime %)))]
```

Comment:
Yuck, I hate this nested closure trick 😄 

How about making a private helper for both steps:

```clojure
(defn- parse-offset-datetime [s] (.toOffsetDateTime ^ZonedDateTime (u.date/parse s default-timezone)))
```

Note we should also provide a default timezone to ensure we never get back a `LocalDate` and violate the type hint.

Reply:
> I hate this nested closure trick 

any reasons why? I get that it's a bit weird but I don't think it's bad in anyway?

Reply:
I'm happy to move this to a helper function but the real reason I do it like that is because of the some-> macros, which shorten things quite a bit.